geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java

```diff
@@ -764,6 +772,16 @@ protected void doHandshake() {
     }
   }
 
+  private boolean incrementConnectionsProcessing() {
+    if (serverConnectionCollection.isTerminating) {
+      serverConnectionCollection.connectionsProcessing.incrementAndGet();
```

Comment:
The old logic was if terminating then don't increment.
But this new method only increments if terminating.
Also I think you want to return false if terminating and true if you increment.

Reply:
Thank you, I meant to have a ! in that if statement